### PR TITLE
[TEXT ANALYTICS] Split up tests into endpoints, increase test coverage

### LINKED ISSUE #9445: [textanalytics] Improve test coverage
Add more tests / see where more coverage needed

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #9445

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_detect_language.py [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+        docs = [u"hello world"] * 1001
+        try:
+            response = text_analytics.detect_language(docs)
+        except HttpResponseError as err:
+            self.assertEqual(err.error_code, "InvalidDocumentBatch")
+            self.assertIsNotNone(err.message)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @pytest.mark.skip(reason="Service bug returns invalidDocument here. Unskip after v3.0-preview.2")
+    def test_text_analytics_error_bad_model_version(self, resource_group, location, text_analytics_account, text_analytics_account_key):
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_detect_language.py]: Name seems wrong here. We're testing for invalid country hint

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+        except AttributeError as default_behavior:
+            self.assertEqual(
+                default_behavior.args[0],
+                '\'DocumentError\' object has no attribute \'attribute_not_on_result_or_error\''
+            )
+
+    @GlobalTextAnalyticsAccountPreparer()
+    def test_text_analytics_error_bad_model_version(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+        text = ""
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py]: The too large document can be removed for this test.

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+    @GlobalTextAnalyticsAccountPreparer()
+    def test_text_analytics_error_missing_input_records(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+        text = ""
+        for _ in range(5121):
+            text += "x"
+
+        docs = [{"id": "1", "text": ""},
+                {"id": "2", "language": "english", "text": "I did not like the hotel we stayed it."},
+                {"id": "3", "text": text}]
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py]: Same here, these lines can be removed

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py [hunk 4]
```diff
@@ -0,0 +1,545 @@
...
+    @GlobalTextAnalyticsAccountPreparer()
+    def test_text_analytics_error_duplicate_ids(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+        text = ""
+        for _ in range(5121):
+            text += "x"
+
+        docs = [{"id": "1", "text": ""},
+                {"id": "2", "language": "english", "text": "I did not like the hotel we stayed it."},
+                {"id": "3", "text": text}]
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py]: Can remove these lines

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py [hunk 5]
```diff
@@ -0,0 +1,545 @@
...
+    @GlobalTextAnalyticsAccountPreparer()
+    def test_text_analytics_error_batch_size_over_limit(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+        text = ""
+        for _ in range(5121):
+            text += "x"
+
+        docs = [{"id": "1", "text": ""},
+                {"id": "2", "language": "english", "text": "I did not like the hotel we stayed it."},
+                {"id": "3", "text": text}]
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py]: Remove these lines

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py [hunk 6]
```diff
@@ -0,0 +1,545 @@
...
+        docs = [u"hello world"] * 1001
+        try:
+            response = text_analytics.analyze_sentiment(docs)
+        except HttpResponseError as err:
+            self.assertEqual(err.error_code, "InvalidDocumentBatch")
+            self.assertIsNotNone(err.message)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @pytest.mark.skip(reason="Service bug returns invalidDocument here. Unskip after v3.0-preview.2")
+    def test_text_analytics_error_bad_model_version(self, resource_group, location, text_analytics_account, text_analytics_account_key):
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py]: This looks like it's testing country hint, but is titled bad model version? We already tested a bad model version above, and country hint doesn't apply to the sentiment endpoint so let's remove this test.

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py [hunk 7]
```diff
@@ -0,0 +1,545 @@
...
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+
+        docs = [{"id": "1", "country_hint": "United States", "text": "hello world"}]
+
+        response = text_analytics.analyze_sentiment(docs)
+        self.assertEqual(response[0].error.code, "invalidCountryHint")
+        self.assertIsNotNone(response[0].error.message)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    def test_language_kwarg_spanish(self, resource_group, location, text_analytics_account, text_analytics_account_key):
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py]: This test and the test below it have the same name. We just need one, this test just aims to make sure that the kwargs passed get used in the request

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py [hunk 8]
```diff
@@ -0,0 +1,593 @@
...
+                '\'DocumentError\' object has no attribute \'attribute_not_on_result_or_error\''
+            )
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @AsyncTextAnalyticsTest.await_prepared_test
+    async def test_text_analytics_error_bad_model_version(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+        text = ""
+        for _ in range(5121):
+            text += "x"
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py]: can remove this if just checking bad model version

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py [hunk 9]
```diff
@@ -0,0 +1,593 @@
...
+    @AsyncTextAnalyticsTest.await_prepared_test
+    async def test_text_analytics_error_missing_input_records(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+        text = ""
+        for _ in range(5121):
+            text += "x"
+
+        docs = [{"id": "1", "text": ""},
+                {"id": "2", "language": "english", "text": "I did not like the hotel we stayed it."},
+                {"id": "3", "text": text}]
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py]: can remove

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py [hunk 10]
```diff
@@ -0,0 +1,593 @@
...
+    async def test_text_analytics_error_duplicate_ids(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+        text = ""
+        for _ in range(5121):
+            text += "x"
+
+        docs = [{"id": "1", "text": ""},
+                {"id": "2", "language": "english", "text": "I did not like the hotel we stayed it."},
+                {"id": "3", "text": text}]
+
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py]: can remove

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py [hunk 11]
```diff
@@ -0,0 +1,593 @@
...
+    @AsyncTextAnalyticsTest.await_prepared_test
+    async def test_text_analytics_error_batch_size_over_limit(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+        text = ""
+        for _ in range(5121):
+            text += "x"
+
+        docs = [{"id": "1", "text": ""},
+                {"id": "2", "language": "english", "text": "I did not like the hotel we stayed it."},
+                {"id": "3", "text": text}]
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py]: This copy/paste error can be removed in all test files where it's not used.

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py [hunk 12]
```diff
@@ -0,0 +1,593 @@
...
+        try:
+            response = await text_analytics.analyze_sentiment(docs)
+        except HttpResponseError as err:
+            self.assertEqual(err.error_code, "InvalidDocumentBatch")
+            self.assertIsNotNone(err.message)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @AsyncTextAnalyticsTest.await_prepared_test
+    @pytest.mark.skip(reason="Service bug returns invalidDocument here. Unskip after v3.0-preview.2")
+    async def test_text_analytics_error_bad_model_version(self, resource_group, location, text_analytics_account, text_analytics_account_key):
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py]: Remove this test

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py [hunk 13]
```diff
@@ -0,0 +1,593 @@
...
+
+        docs = [{"id": "1", "country_hint": "United States", "text": "hello world"}]
+
+        response = await text_analytics.analyze_sentiment(docs)
+        self.assertEqual(response[0].error.code, "invalidCountryHint")
+        self.assertIsNotNone(response[0].error.message)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @AsyncTextAnalyticsTest.await_prepared_test
+    async def test_language_kwarg_spanish(self, resource_group, location, text_analytics_account, text_analytics_account_key):
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment_async.py]: remove duplicate test below this one

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_detect_language.py [hunk 14]
```diff
@@ -0,0 +1,569 @@
...
+        except AttributeError as default_behavior:
+            self.assertEqual(
+                default_behavior.args[0],
+                '\'DocumentError\' object has no attribute \'attribute_not_on_result_or_error\''
+            )
+
+    @GlobalTextAnalyticsAccountPreparer()
+    def test_text_analytics_error_bad_model_version(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+        text = ""
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_detect_language.py]: Remove the code that prepares the large doc for the series of document error tests (applies to async file as well)

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_detect_language_async.py [hunk 15]
```diff
@@ -0,0 +1,616 @@
...
+        try:
+            response = await text_analytics.detect_language(docs)
+        except HttpResponseError as err:
+            self.assertEqual(err.error_code, "InvalidDocumentBatch")
+            self.assertIsNotNone(err.message)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @AsyncTextAnalyticsTest.await_prepared_test
+    @pytest.mark.skip(reason="Service bug returns invalidDocument here. Unskip after v3.0-preview.2")
+    async def test_text_analytics_error_bad_model_version(self, resource_group, location, text_analytics_account, text_analytics_account_key):
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_detect_language_async.py]: change test name to testing invalid country hint

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_extract_key_phrases.py [hunk 16]
```diff
@@ -0,0 +1,516 @@
...
+        docs = [u"hello world"] * 1001
+        try:
+            response = text_analytics.extract_key_phrases(docs)
+        except HttpResponseError as err:
+            self.assertEqual(err.error_code, "InvalidDocumentBatch")
+            self.assertIsNotNone(err.message)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @pytest.mark.skip(reason="Service bug returns invalidDocument here. Unskip after v3.0-preview.2")
+    def test_text_analytics_error_bad_model_version(self, resource_group, location, text_analytics_account, text_analytics_account_key):
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_extract_key_phrases.py]: remove test, don't need to test country hint for key phrases endpoint

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_extract_key_phrases.py [hunk 17]
```diff
@@ -0,0 +1,516 @@
...
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+
+        docs = [{"id": "1", "country_hint": "United States", "text": "hello world"}]
+
+        response = text_analytics.extract_key_phrases(docs)
+        self.assertEqual(response[0].error.code, "invalidCountryHint")
+        self.assertIsNotNone(response[0].error.message)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    def test_language_kwarg_spanish(self, resource_group, location, text_analytics_account, text_analytics_account_key):
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_extract_key_phrases.py]: remove duplicate test

**Comment by kristapratico** [comment]
Some copy/paste errors across test files, but otherwise looks good. thanks!

File: sdk/textanalytics/azure-ai-textanalytics/tests/test_recognize_entities_async.py [hunk 18]
```diff
@@ -0,0 +1,526 @@
...
+                self.assertIsNotNone(entity.grapheme_offset)
+                self.assertIsNotNone(entity.grapheme_length)
+                self.assertIsNotNone(entity.score)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @AsyncTextAnalyticsTest.await_prepared_test
+    async def test_length_with_emoji(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+
+        result = await text_analytics.recognize_pii_entities(["👩 SSN: 123-12-1234"])
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/test_recognize_entities_async.py]: This test can just go in recognize_pii_entities test file

**Comment by kristapratico** [comment]
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/textanalytics/azure-ai-textanalytics/tests/test_batch.py#L545

File: sdk/textanalytics/azure-ai-textanalytics/tests/recordings/test_auth.test_active_directory_auth.yaml [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+      - '354'
+      Content-Type:
+      - application/json; charset=utf-8
+      User-Agent:
+      - azsdk-python-azure-ai-textanalytics/1.0.0b3 Python/3.6.6 (Windows-10-10.0.18362-SP0)
+    method: POST
+    uri: https://python-textanalytics.cognitiveservices.azure.com/text/analytics/v3.0-preview.1/languages?showStats=false
+  response:
+    body:
+      string: '{ "error": { "code": "PermissionDenied", "message": "Principal does
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/tests/recordings/test_auth.test_active_directory_auth.yaml]: Re-record this. It looks like the failing auth recording got commited

**Pull request merged**